var/spack/repos/builtin/packages/ufs-weather-model/package.py

```diff
@@ -17,68 +17,160 @@ class UfsWeatherModel(CMakePackage):
     url = "https://github.com/ufs-community/ufs-weather-model/archive/refs/tags/ufs-v1.1.0.tar.gz"
     git = "https://github.com/ufs-community/ufs-weather-model.git"
 
-    maintainers("t-brown")
+    maintainers("AlexanderRichert-NOAA")
 
-    version(
-        "2.0.0",
-        tag="ufs-v2.0.0",
-        commit="e3cb92f1cd8941c019ee5ef7da5c9aef67d55cf8",
-        submodules=True,
-    )
-    version(
-        "1.1.0",
-        tag="ufs-v1.1.0",
-        commit="5bea16b6d41d810dc2e45cba0fa3841f45ea7c7a",
-        submodules=True,
-    )
+    version("develop", branch="develop", submodules=True)
+    version("2.0.0", tag="ufs-v2.0.0", submodules=True)
+    version("1.1.0", tag="ufs-v1.1.0", submodules=True)
 
+    variant("mpi", default=True, description="Enable MPI")
     variant(
         "32bit", default=True, description="Enable 32-bit single precision arithmetic in dycore"
     )
-    variant("avx2", default=False, description="Enable AVX2 instructions")
     variant(
-        "ccpp", default=True, description="Enable the Common Community Physics Package (CCPP))"
+        "ccpp_32bit",
+        default=False,
+        description="Enable CCPP_32BIT (single precision arithmetic in slow physics)",
+    )
+    variant("avx2", default=True, description="Enable AVX2 instruction set")
```

Comment:
This should probably be controlled by whether `avx2` is in the target, not by the variant. As currently written this could concretize for avx2 and a target that doesn't support it.

IIRC you can make queries like `if "avx2" in self.target"` to check for things like this.

Reply:
Is there a way to create a conflict that encapsulates that? Since it's a cmake option, I think it would make sense to still allow users to disable it for debugging purposes.

Reply:
You could set it up with a when-clause, but it would be a bit clunky

Reply:
Would this basically involve creating an entry for each CPU architecture (`when="target=x86_64_v3"`, `when="target=haswell"`, etc.)?

Reply:
You can do ranges of targets, so you could say when="target=:haswell" or whatever the appropriate target is. But you'd have to have a check for the intel cpus, a check for the amd cpus, and checks for ppc and arm.

Much better to just trigger it by whether the arch supports it and build for the older arch if you don't want to use it.

Reply:
Since those settings get used for specific sites anyway (at those sites ufs wm will not be installed through spack), I'm setting those cmake options to OFF and removing the variants so that the target settings will be set purely through spack.